### PR TITLE
Feature/auth

### PR DESCRIPTION
**Pull request by AlexYalovoy**

### CODE REVIEW

File: src/components/Header/index.js [hunk 1]
```diff
@@ -25,74 +27,93 @@ const style = {
...
-                  </Button>
-                </Link>
-              </Fragment>
-            )}
-            <Link to="/profile">
-              <Button className="rightButton">
-                <Person />
-                <span className="text">{login}</span>
+const Header = ({ classes, isLoggedIn, role, onLogOut }) => {
+  const LogOutHandler = () => onLogOut();
```
Comment (motifej) [inline src/components/Header/index.js]: explain this line

File: src/components/Header/index.js [hunk 2]
```diff
@@ -25,74 +27,93 @@ const style = {
...
+          ) : (
+            <div className={classes.brand}>
+              <Typography variant="title" className={classes.title}>
+                LOGO
+              </Typography>
+            </div>
+          )}
+          {isLoggedIn ? (
+            <Fragment>
+              {role === 'admin' ? (
```
Comment (motifej) [inline src/components/Header/index.js]: admin should be a const
const ROLE = {
ADMIN: 'admin'
}

File: src/sagas/signInSaga/index.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+import { signInError, signInSuccess } from 'actions/auth';
+import { saveProfile } from 'actions/user';
+import { signIn } from 'api/auth';
+import { getUserfromDB } from 'api/database';
+import { SIGNIN } from 'actions/types';
+
+function* SignIn({ payload: { email, password } }) {
+  try {
+    const response = yield call(signIn, email, password);
+    const user = yield call(getUserfromDB, response.user.uid);
```
Comment (motifej) [inline src/sagas/signInSaga/index.js]: you are do more then one operation in this saga, split logic

File: src/components/Header/index.js [hunk 4]
```diff
@@ -1,16 +1,19 @@
 import React, { Fragment } from 'react';
 import { Link } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { compose } from 'recompose';
```
[the remark this answers is not in the document]
Comment (AlexYalovoy) [reply src/components/Header/index.js]: Good call, we can use this function and not to install recompose.
Let's try to use it in future)

File: src/reducers/auth.reducer.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+const initialState = {
+  uid: null,
+  isLoggedIn: false,
+  error: null,
+};
+
+export default combineEvents(
+  {
+    [SIGNIN]: () => ({}),
```
Comment (motifej) [inline src/reducers/auth.reducer.js]: remove it, add only it nessesary

File: src/sagas/signInSaga/index.js [hunk 6]
```diff
@@ -0,0 +1,17 @@
+import { call, put, takeEvery } from 'redux-saga/effects';
+import { signInError, signInSuccess } from 'actions/auth';
+import { signIn } from 'api/auth';
+import { SIGNIN } from 'actions/types';
+
+function* SignIn({ payload: { email, password } }) {
+  try {
```
Comment (motifej) [inline src/sagas/signInSaga/index.js]: add action for start

File: src/sagas/signUpSaga/index.js [hunk 7]
```diff
@@ -1,20 +1,25 @@
...
+import { signUpError, signUpSuccess } from 'actions/auth';
+import { register } from 'api/auth';
 
-function* SignUp(action) {
-  const { email, password } = action.payload;
+function* SignUp({ payload: { email, password, firstName, lastName, phone } }) {
   try {
-    yield call(register, email, password);
-    yield put(signUpSuccess());
+    const response = yield call(register, email, password);
```
Comment (motifej) [inline src/sagas/signUpSaga/index.js]: add action for start

**Pull request merged**